src/protocol/ipa/mod.rs

```diff
@@ -332,4 +341,45 @@ mod tests {
             assert_eq!(*expected, [result[0], result[1]]);
         }
     }
+
+    #[tokio::test]
+    #[allow(clippy::missing_panics_doc)]
+    pub async fn random_ipa_no_result_check() {
+        const BATCHSIZE: u64 = 100;
+        const PER_USER_CAP: u32 = 10;
+        const MAX_BREAKDOWN_KEY: u128 = 8;
+        const MAX_TRIGGER_VALUE: u128 = 5;
+        let matchkeys_upto: u64 = BATCHSIZE / 10;
+
+        let world = TestWorld::new(QueryId);
+        let mut rng = thread_rng();
+
+        let mut records: Vec<IPAInputTestRow> = Vec::new();
+
+        for _ in 0..BATCHSIZE {
+            let is_trigger_bit = u128::from(rng.gen::<bool>());
+            let test_row = IPAInputTestRow {
+                match_key: rng.gen_range(0..matchkeys_upto),
+                is_trigger_bit,
+                breakdown_key: match is_trigger_bit {
+                    0 => rng.gen_range(0..MAX_BREAKDOWN_KEY), // Breakdown key is only found in source events
+                    1_u128..=u128::MAX => 0,
+                },
+                trigger_value: is_trigger_bit * rng.gen_range(1..MAX_TRIGGER_VALUE), // Trigger value is only found in trigger events
```

Comment:
This is duplicate code. Can you make a `IPAInputTestRow::random(&mut rng)` method for this?

Reply:
Can! But I would still need to expose it in two places (since this is test code). Unless @martinthomson/ @akoshelev can suggest some tips

Reply:
Actually, I am moving it to test-fixture and that should be made accessible then